crates/sui-rpc-benchmark/src/direct/query_executor.rs

```diff
@@ -0,0 +1,277 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use tokio_postgres::{types::ToSql, types::Type, Client, Row};
+
+use crate::direct::query_generator::BenchmarkQuery;
+
+pub struct QueryExecutor {
+    db_client: Client,
+    enriched_benchmark_queries: Vec<EnrichedBenchmarkQuery>,
+}
+
+#[derive(Debug)]
+pub struct EnrichedBenchmarkQuery {
+    pub query: BenchmarkQuery,
+    pub rows: Vec<Row>,
+}
+
+impl QueryExecutor {
+    pub async fn new(
+        db_url: &str,
+        benchmark_queries: Vec<BenchmarkQuery>,
+    ) -> Result<Self, anyhow::Error> {
+        let (client, connection) = tokio_postgres::connect(db_url, tokio_postgres::NoTls).await?;
+        tokio::spawn(async move {
+            if let Err(e) = connection.await {
+                eprintln!("connection error: {}", e);
+            }
+        });
+
+        let mut executor = Self {
+            db_client: client,
```

Comment:
Does this mean all queries will share the same connection? Could that become a bottleneck?

Reply:
for now yes, but I see your point and indeed planned to handle threads and connections as a follow to avoid further bloating this pr.